src/Symfony/Component/DependencyInjection/ContainerInterface.php

```diff
@@ -21,7 +22,7 @@
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
  */
-interface ContainerInterface
+interface ContainerInterface extends PsrContainerInterface
```

Comment:
Because PHP isn't aware of class hierarchy at compile time so it can't allow changing the type declaration.